public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -67,44 +71,33 @@ export class PanelQueryRunner {
    * the results will be immediatly passed to the observer
    */
   subscribe(observer: PartialObserver<PanelData>, format = PanelQueryRunnerFormat.frames): Unsubscribable {
-    if (format === PanelQueryRunnerFormat.legacy) {
-      this.state.sendLegacy = true;
-    } else if (format === PanelQueryRunnerFormat.both) {
-      this.state.sendFrames = true;
-      this.state.sendLegacy = true;
-    } else {
-      this.state.sendFrames = true;
-    }
-
-    // Send the last result
-    if (this.state.isStarted()) {
-      observer.next(this.state.getDataAfterCheckingFormats());
-    }
+    return this.subject.subscribe({
```

Comment:
Not sure if consciously or not but we are ignoring any errors and completions here. I think this would be better as `return this.subject.pipe(map(/* do the mapping*/))`. The function will change to `getObservable(format): Observable`. That way errors should be propagated upstream.

Reply:
subscriber does not care about completions or errors (via the Observer functions with that name) as state & error is part of the PanelData, and error handling is part of runRequest:   https://github.com/grafana/grafana/blob/query-observable/public/app/features/dashboard/state/runRequest.ts#L102

We could reconsider how query errors are returned and use the error Observer callback instead, easy to change to later as it would not impact anything in data source interface or usage patterns.